src/domain/session/room/timeline/tiles/GapTile.js

```diff
@@ -42,11 +43,19 @@ export class GapTile extends SimpleTile {
                 this._loading = false;
                 this.emitChange("isLoading");
             }
+                return true;
         }
+        return false;
     }
 
-    notifyVisible() {
-        this.fill();
+    async notifyVisible() {
+        let depth = 0;
+        let result;
+        do {
+            result = await this.fill();
+            depth = depth + 1;
+        } while (depth < 10 && !this._siblingChanged && result && !this.isDisposed)
```

Comment:
We check `isDisposed` here so that we don't start an extra gap fill before the entry update makes its way to GapTile.